### ISSUE TITLE
RqMultiPart.Fake fails on more then one part

### ISSUE THREAD

**Issue by jhyle** [yegor256/takes#577]
Please see https://github.com/yegor256/netbout/pull/979. The TkAttachTest::takeAttachementNameFromField constructs a RqMultiPart.Fake request with two form-data parts. It fails with the exception

```
org.takes.HttpException: [400] header "Content-Disposition" is mandatory
    at org.takes.rq.RqHeaders$Smart.single(RqHeaders.java:206)
    at org.takes.rq.RqMultipart$Base.asMap(RqMultipart.java:313)
    at org.takes.rq.RqMultipart$Base.buildRequests(RqMultipart.java:224)
    at org.takes.rq.RqMultipart$Base.<init>(RqMultipart.java:140)
    at org.takes.rq.RqMultipart$Fake.<init>(RqMultipart.java:413)
    at com.netbout.rest.bout.TkAttachTest.takettachementNameFromField(TkAttachTest.java:214)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:318)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
```

As `Content-Disposition` is given for both parts I suspect that RqMultiPart.Fake does not handle this correctly.

**Comment by davvd** [comment]
@yegor256 please do something about this issue

**Comment by dmzaytsev** [comment]
@yegor256 please do something about this issue

**Comment by krzyk** [comment]
@yegor256 could you validate if this is a real bug?

**Comment by yegor256** [comment]
@jhyle can you please post a unit test here, to reproduce the bug?

**Comment by yegor256** [comment]
@jhyle is it still a problem or I can close the ticket?

**Comment by paulodamaso** [comment]
@0crat in

**Comment by 0crat** [comment]
The job [#577](https://github.com/yegor256/takes/issues/577) assigned to @robsonataide[/z](https://www.0crat.com/u/robsonataide), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/73698007-e289-4e04-995e-023cd79cbdf5); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@robsonataide[/z](https://www.0crat.com/u/robsonataide) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @robsonataide[/z](https://www.0crat.com/u/robsonataide) resigned from [#577](https://github.com/yegor256/takes/issues/577), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -30 point(s) just awarded to @robsonataide[/z](https://www.0crat.com/u/robsonataide)

**Comment by 0crat** [comment]
The job [#577](https://github.com/yegor256/takes/issues/577) assigned to @serranya[/z](https://www.0crat.com/u/serranya), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/c6866acc-d4a5-41c1-b978-d5a5f006bf28); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0pdd** [comment]
@jhyle the puzzle [#950](https://github.com/yegor256/takes/issues/950) is still not solved.

**Comment by Serranya** [comment]
@jhyle I added an Test that repduces the issue in https://github.com/yegor256/takes/pull/947. Actually fixing the issue is tracked in https://github.com/yegor256/takes/issues/950. Please close this issue so that only https://github.com/yegor256/takes/issues/950 is open.

**Comment by Serranya** [comment]
@0crat wait for an answer from @jhyle

**Comment by 0crat** [comment]
> @0crat wait for an answer from @jhyle  ([here](https://github.com/yegor256/takes/issues/577#issuecomment-455867829))

@Serranya The impediment for [#577](https://github.com/yegor256/takes/issues/577) was registered successfully by @serranya[/z](https://www.0crat.com/u/serranya)

**Comment by Serranya** [comment]
@paulodamaso Hi, @jhyle did not answer in the last 8 days. Can you please close the issue?

**Comment by paulodamaso** [comment]
@Serranya Thanks!

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @serranya[/z](https://www.0crat.com/u/serranya), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#577](https://github.com/yegor256/takes/issues/577) is now out of scope

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
> @0crat quality good ([here](https://github.com/yegor256/takes/issues/577#issuecomment-458092366))

@ypshenychka The project doesn't have enough funds, can't make a payment

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @serranya[/z](https://www.0crat.com/u/serranya)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0pdd** [comment]
@jhyle the puzzle [#1095](https://github.com/yegor256/takes/issues/1095) is still not solved; solved: [#950](https://github.com/yegor256/takes/issues/950).

**Comment by 0pdd** [comment]
@jhyle all 2 puzzles are solved here: [#1095](https://github.com/yegor256/takes/issues/1095), [#950](https://github.com/yegor256/takes/issues/950).
